### PR TITLE
SCIDE: Separate server actions into Server menu

### LINKED ISSUE #2852: Language menu is too big
Low priority, not for 3.9, but... here's the Language menu with XFCE's default system font on a 768-pixel-high screen (typical for laptops):

language-menu

It's not pretty.

Concrete suggestion: Many of these options are about the Server, not the Language, so we could have a separate Server menu.

### PR DESCRIPTION
**Pull request by kisielk**
Fixes #2852

### CODE REVIEW

File: editors/sc-ide/widgets/main_window.cpp [hunk 1]
```diff
@@ -654,6 +654,19 @@ void MainWindow::createMenus()
...
+    menu->addAction( mMain->scProcess()->action(ScIDE::ScProcess::StopMain) );
+    menu->addSeparator();
+    menu->addAction( mActions[LookupImplementationForCursor] );
+    menu->addAction( mActions[LookupImplementation] );
+    menu->addAction( mActions[LookupReferencesForCursor] );
+    menu->addAction( mActions[LookupReferences] );
+
+    menuBar->addMenu(menu);
+
+    menu = new QMenu(tr("Se&rver"), this);
```
Comment (brianlheim) [inline editors/sc-ide/widgets/main_window.cpp]: Shouldn't this be `"&Server"`?

Comment (kisielk) [inline editors/sc-ide/widgets/main_window.cpp]: S is already the shortcut for the Session menu, so I had to pick something else.

**Comment by brianlheim** [comment]
Tested and works on macOS

**Comment by kisielk** [comment]
I guess I should mentioned I developed / tested this on macOS as well. Needs some testing on Linux and Windows. I don't have the dev environment set up on either just yet.

**Comment by brianlheim** [comment]
np. I get the impression that Qt stuff is usually good cross-platform, but I don't know a lot about it so I'm being cautious. I'm about to build this on Windows actually.

**Comment by brianlheim** [comment]
works on windows!

**Comment by patrickdupuis** [comment]
Looks good on Linux over here 👍 Thanks!

**Pull request merged**